From abb99e0703e9c2a672af712c44b451ce3fa0dd0e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 6 May 2018 12:23:38 +0100
Subject: [PATCH 3/3] Add evil `exit 0` script for Dell bluetooth

This is a quick hack/workaround until the upstream kernel bug is fixed
with Dell switching reporting itself as broken, despite working. In turn
this will alleviate the 100% CPU usage issue seen with udevd.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile.tools             | 1 +
 tools/hid2hci-pass-through | 3 +++
 tools/hid2hci.rules        | 2 +-
 3 files changed, 5 insertions(+), 1 deletion(-)
 create mode 100755 tools/hid2hci-pass-through

diff --git a/Makefile.tools b/Makefile.tools
index b7b4225..a01cbcb 100644
--- a/Makefile.tools
+++ b/Makefile.tools
@@ -365,6 +365,7 @@ if HID2HCI
 udevdir = @UDEV_DIR@
 
 udev_PROGRAMS = tools/hid2hci
+udev_SCRIPTS = tools/hid2hci-pass-through
 
 tools_hid2hci_LDADD = @UDEV_LIBS@
 
diff --git a/tools/hid2hci-pass-through b/tools/hid2hci-pass-through
new file mode 100755
index 0000000..1fca8ab
--- /dev/null
+++ b/tools/hid2hci-pass-through
@@ -0,0 +1,3 @@
+#!/bin/bash
+/usr/lib/udev/hid2hci $*
+exit 0
diff --git a/tools/hid2hci.rules b/tools/hid2hci.rules
index db6bb03..b228c88 100644
--- a/tools/hid2hci.rules
+++ b/tools/hid2hci.rules
@@ -8,7 +8,7 @@ SUBSYSTEM!="usb*", GOTO="hid2hci_end"
 # Known supported devices: 413c:8154, 413c:8158, 413c:8162
 ATTR{bInterfaceClass}=="03", ATTR{bInterfaceSubClass}=="01", ATTR{bInterfaceProtocol}=="02", \
   ATTRS{bDeviceClass}=="00", ATTRS{idVendor}=="413c", ATTRS{bmAttributes}=="e0", \
-  RUN+="hid2hci --method=dell --devpath=%p", ENV{HID2HCI_SWITCH}="1"
+  RUN+="hid2hci-pass-through --method=dell --devpath=%p", ENV{HID2HCI_SWITCH}="1"
 
 # Logitech devices
 KERNEL=="hiddev*", ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c70[345abce]|c71[34bc]", \
-- 
2.16.3

